src/solace_ai_connector/components/general/llm/langchain/langchain_embeddings.py

```diff
@@ -66,13 +66,28 @@ def __init__(self, **kwargs):
         super().__init__(info, **kwargs)
 
     def invoke(self, message, data):
-        text = data["text"]
+        items = data["items"]
         embedding_type = data.get("type", "document")
 
-        embeddings = None
+        items = [items] if type(items) != list else items
```

Comment:
[SUG]: You can improve the validation step before getting an error in the next steps. Maybe check None and/or check the content of items based on types (document, image or query).